### PR TITLE
PR for new test case adde to verify https://github.com/couchbase/couchbase-lite-java-core/issues/1413

### PR DESCRIPTION
**Pull request by hrajput89**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Added a new test case for https://github.com/couchbase/couchbase-lite-java-core/issues/1413

### CODE REVIEW

File: testsuites/listener/shared/client_sg/test_design_docs_update.py [hunk 1]
```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
...
+
+    # Design doc to to fetch doc._id, doc._rev for docs with no content
+    view = """{
+    "language" : "javascript",
+    "views" : {
+        "content_view" : {
+            "map" : "function(doc, meta) { if (!(doc.content)) { emit(doc._id, doc._rev); } }"
+        }
+    }
+}"""
```
Comment (sridevi-15) [inline testsuites/listener/shared/client_sg/test_design_docs_update.py]: can you intend appropriately?

File: testsuites/listener/shared/client_sg/test_design_docs_update.py [hunk 2]
```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
...
+
+    # creating a new Desing doc  with same name as previous design doc
+    log_info("Creating a new design doc to query docs with no content with same name as previous one")
+    client.add_design_doc(url=ls_url, db=ls_db, name=d_doc_name, doc=view)
+    dd_rev_new = client.get_design_doc_rev(url=ls_url, db=ls_db, name=d_doc_name)
+#     assert dd_rev_new != dd_rev and dd_rev_new.startswith("1-")
```
Comment (sridevi-15) [inline testsuites/listener/shared/client_sg/test_design_docs_update.py]: can you remove this comment if it is not necessary

File: testsuites/listener/shared/client_sg/test_design_docs_update.py [hunk 3]
```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
...
     client.verify_view_row_num(view_response=content_view_rows_2, expected_num_rows=5)
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.syncgateway
+@pytest.mark.attachments
+def test_design_doc_delete(setup_client_syncgateway_test):
+    """
+    Ref: https://issues.couchbase.com/browse/CBSE-4650
```
Comment (sridevi-15) [inline testsuites/listener/shared/client_sg/test_design_docs_update.py]: can you add @summary tag

File: testsuites/listener/shared/client_sg/test_design_docs_update.py [hunk 4]
```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
 
     content_view_rows_2 = client.get_view(url=ls_url, db=ls_db, design_doc_name=d_doc_name, view_name="content_view")
     client.verify_view_row_num(view_response=content_view_rows_2, expected_num_rows=5)
+
+
+@pytest.mark.sanity
```
Comment (sridevi-15) [inline testsuites/listener/shared/client_sg/test_design_docs_update.py]: do you want to mark this as sanity? I guess it is not required

Comment (hrajput89) [reply testsuites/listener/shared/client_sg/test_design_docs_update.py]: removed.

File: testsuites/listener/shared/client_sg/test_design_docs_update.py [hunk 5]
```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
...
+
+    # Design doc to to fetch doc._id, doc._rev for docs with content
+    view = """{
+    "language" : "javascript",
+    "views" : {
+        "content_view" : {
+            "map" : "function(doc, meta) { if (doc.content) { emit(doc._id, doc._rev); } }"
+        }
+    }
+}"""
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply testsuites/listener/shared/client_sg/test_design_docs_update.py]: done

File: testsuites/listener/shared/client_sg/test_design_docs_update.py (hunk 1 above)
Comment (hrajput89) [reply testsuites/listener/shared/client_sg/test_design_docs_update.py]: done

File: testsuites/listener/shared/client_sg/test_design_docs_update.py (hunk 2 above)
Comment (hrajput89) [reply testsuites/listener/shared/client_sg/test_design_docs_update.py]: removed.

File: testsuites/listener/shared/client_sg/test_design_docs_update.py (hunk 3 above)
Comment (hrajput89) [reply testsuites/listener/shared/client_sg/test_design_docs_update.py]: done

**Comment by hrajput89** [comment]
Made changes according to @sridevi-15 comments

**Pull request merged**